lib/src/native/realm_core.dart

```diff
@@ -196,6 +199,29 @@ class _RealmCore {
     });
   }
 
+  static int initial_data_callback(Object config, Pointer<shared_realm> realmHandle) {
+    try {
+      if (config is Configuration) {
+        final realm = RealmInternal.getUnowned(config, RealmHandle._unowned(realmHandle));
+        config.initialDataCallback!(realm);
+        return TRUE;
+      }
+    } catch (ex) {
+      // TODO: this should propagate the error to Core: https://github.com/realm/realm-core/issues/5366
+      print(ex);
```

Comment:
If we bring down the process, there'll be no way to write a test for this case. Additionally, it's not like we can't tell Core that an exception happened - it's just that we can't tell the user what the precise error is. Right now, it'll just throw a generic error from the realm ctor. This is covered by this test case:

https://github.com/realm/realm-dart/blob/7b04d9d0d1b6312f285b79e1cb8a324ad5896b66/test/configuration_test.dart#L229-L238

The reason this test is skipped is an unrelated bug in Core that leaves the Realm open when such an error occurs, which then prevents us from deleting it.

Reply:
Still I wonder if we really should be propagating the unhandled error to core. If we throw an `RealmError` here it will mean Dart/Flutter will provide the necessary logging and stacktrace to the developer. Also it will force the developer to think of unhandled errors from such callbacks. Or said differently if we allow the propagation of unhandled errors core will only throw a more specific exception but loosing the stacktrace in the process versus if we throw `RealmError` from the Dart code then the VM will log the exact stacktrace and will allow Isolate unhandled error callback to run. So in a sense telling Core about the exception is hurting the developer experience a bit. Also as a result the realm creation will fail versus if the user is forced to handle the exception on their own inside the callback than it might turn out that an exception happened while initializing the data but still the realm could be created with no initial data. 
Hope it makes sense. I guess we could discuss it more on the round table.

Reply:
Forgot to mention there is a way to test this using `Isolate.spawn`

Reply:
Hm... looks like the ffi will prevent us from bringing down the process here, even if we throw. When we construct the function pointer, we have to provide `exceptionalReturn` argument:

https://github.com/realm/realm-dart/blob/a8502e156635e3d8470270e784315283a160e7b6/lib/src/native/realm_core.dart#L160-L161

My guess is `Pointer.fromFunction` does a try-catch internally and returns the exceptional return value (`FALSE`), even if we don't handle the exception ourselves.

Reply:
Oh, my bad. I have tested it and indeed we can't throw through int callbacks. I guess my suggestion is to just remove the `print` in this case. Later when its possible we could report the exact error to core to be included in the generic error they throw.